src/index.js

```diff
@@ -25,7 +25,7 @@ var enhance = module.exports = function enhance(proto) {
       return originalAddEventListener.call(this, name, originalCallback, optionsOrCapture)
     }
 
-    var callback = originalCallback
+    var callback = typeof originalCallback.handleEvent === 'function' ? originalCallback.handleEvent.bind(originalCallback) : originalCallback
```

Comment:
I'd prefer to flip this ternary; I can't find the spec text to clarify this but the normative behaviour seems to be that passing a function with a `handleEvent` function property, the function is the callback not `handleEvent`. In other words:

```js
const cb = e => console.log('This SHOULD fire',e)
cb.handleEvent = e => console.log('This should NOT fire',e)
document.addEventListener('scroll', cb)
```

Also, interestingly, passing a non-callable object with a non-callable `handleEvent` property seems to silently pass:

```js
const cb = {}
cb.handleEvent = {}
document.addEventListener('scroll', cb) // no errors
```

So perhaps we should stick to that behaviour and change this to the following?

```suggestion
    var callback = typeof originalCallback === 'function' ? originalCallback : typeof originalCallback.handleEvent === 'function' ? originalCallback.handleEvent.bind(originalCallback) : function(){}
```

Reply:
> seems to be that passing a function with a `handleEvent` function property, the function is the callback not `handleEvent`.

👌

> Also, interestingly, passing a non-callable object with a non-callable `handleEvent` property seems to silently pass

Appears that applies to event listeners passed directly too:
```js
document.addEventListener('scroll', {}) // no errors
document.addEventListener('scroll', null) // no errors
document.addEventListener('scroll', new class {}) // no errors
```

So maybe this is the most appropriate change?

```suggestion
var callback = typeof originalCallback !== 'function' && typeof originalCallback.handleEvent === 'function' ? originalCallback.handleEvent.bind(originalCallback) : originalCallback
```

We don't need the final fallback to `function(){}` because the preceding guard clause ensures `originalCallback` is a function or object, which can be passed along as-is. https://github.com/github/eventlistener-polyfill/blob/ee3bd63fca5d5a7e8a7ebeb364b36554507812cf/src/index.js#L23-L25

Reply:
Yes! This change is much better :shipit:!

Reply:
Made that change, added test coverage, squashed and rebased. 🎁